src/daemon/UserConfig.vala

```diff
@@ -23,6 +23,7 @@
 namespace PC.Daemon {
     public class UserConfig : Object {
         public string username { get; set; }
+        public signal void changed (string key);
```

Comment:
Wouldn't it make more sense to have actual GObject properties and use the property notification system instead? (which is more reliable and gives feedback when a property doesn't exist)

Reply:
Indeed but I'm not sure if you can throw errors inside get, set bodies. I will have to test that out.